test/e2e/configuration_test.go

```diff
@@ -29,6 +29,26 @@ import (
 	"github.com/openshift/cluster-image-registry-operator/test/framework"
 )
 
+func TestHTTPSecretDefaulter(t *testing.T) {
+	te := framework.SetupAvailableImageRegistry(t, &imageregistryapiv1.ImageRegistrySpec{
+		ManagementState: operatorapiv1.Managed,
+		Storage: imageregistryapiv1.ImageRegistryConfigStorage{
+			EmptyDir: &imageregistryapiv1.ImageRegistryConfigStorageEmptyDir{},
+		},
+		Replicas: 1,
+	})
+	defer framework.TeardownImageRegistry(te)
+
+	cr, err := te.Client().Configs().Get(context.Background(), defaults.ImageRegistryResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if cr.Spec.HTTPSecret == "" {
+		t.Errorf("got empty spec.httpSecrets, want random string")
+	}
+}
```

Comment:
Could you update this test and set HTTPSecret back to empty and check if it gets populated again?